File: abc/abc148/abc148_c.cs
// https://atcoder.jp/contests/abc148/submissions/9068629

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(long.Parse).ToArray();
            var result = Lcm(input[0], input[1]);
            Console.WriteLine(result);
        }

        public static long Lcm(long a, long b)
        {
            return a * b / Gcd(a, b);
        }

        public static long Gcd(long a, long b)
        {
            if (a < b)
            {
                return Gcd(b, a);
            }
            while (b != 0)
            {
                var r = a % b;
                a = b;
                b = r;
            }
            return a;
        }
    }
}

File: abc/abc148/abc148_c.fs
// https://atcoder.jp/contests/abc148/submissions/12180793

let [| A; B |] = stdin.ReadLine().Split() |> Array.map int64

let rec gcd a b =
    match (a, b) with
    | (x, 0L) -> x
    | (0L, y) -> y
    | (a, b) -> gcd b (a % b)

let lcm a b = a * b / (gcd a b)

lcm A B |> stdout.Write
